packages/core/src/shipping/strategies/braintree/braintree-accelerated-checkout-shipping-strategy.ts

```diff
@@ -176,54 +197,73 @@ export default class BraintreeAcceleratedCheckoutShippingStrategy implements Shi
+            shippingAddresses =
+                this._mapPayPalToBcAddress([profileData.shippingAddress], countries, []) || [];
+            billingAddresses =
+                this._mapPayPalToBcAddress(paypalBillingAddresses, countries, []) || [];
+            instruments = this._mapPayPalToBcInstrument(methodId, [profileData.card]) || [];
+        } else if (isBraintreeConnectProfileData(profileData)) {
+            shippingAddresses =
+                this._mapPayPalToBcAddress(profileData.addresses, countries, profileData.phones) ||
+                [];
+            billingAddresses =
+                this._mapPayPalToBcAddress(paypalBillingAddresses, countries, profileData.phones) ||
+                [];
+            instruments = this._mapPayPalToBcInstrument(methodId, profileData.cards) || [];
+        }
 
-        if (billingAddresses.length > 0) {
-            await this._store.dispatch(
-                this._billingAddressActionCreator.updateAddress({
-                    ...billingAddresses[0],
-                    id: String(billingAddresses[0].id),
-                }),
+        if (shippingAddresses && billingAddresses) {
+            const addresses = this._mergeShippingAndBillingAddresses(
+                shippingAddresses,
+                billingAddresses,
             );
-        }
 
-        if (shippingAddresses.length > 0 && cart.lineItems.physicalItems.length > 0) {
             await this._store.dispatch(
-                this._consignmentActionCreator.updateAddress(shippingAddresses[0]),
+                this._paymentProviderCustomerActionCreator.updatePaymentProviderCustomer({
+                    authenticationState,
+                    addresses,
+                    instruments,
+                }),
             );
+
+            if (billingAddresses.length > 0) {
+                await this._store.dispatch(
+                    this._billingAddressActionCreator.updateAddress({
+                        ...billingAddresses[0],
+                        id: String(billingAddresses[0].id),
+                    }),
+                );
+            }
+
+            if (shippingAddresses.length > 0 && cart.lineItems.physicalItems.length > 0) {
+                await this._store.dispatch(
+                    this._consignmentActionCreator.updateAddress(shippingAddresses[0]),
+                );
+            }
         }
     }
 
     private _getPayPalBillingAddresses(
-        profileData?: BraintreeConnectProfileData,
-    ): BraintreeConnectAddress[] | undefined {
-        const { cards, name } = profileData || {};
-
-        if (!cards?.length) {
-            return;
+        profileData?: BraintreeConnectProfileData | BraintreeFastlaneProfileData,
+    ): BraintreeFastlaneAddress[] | undefined {
+        let cards;
+        const { name } = profileData || {};
+
+        if (profileData) {
+            if (isBraintreeFastlaneProfileData(profileData)) {
```

Comment:
Try to combine checking `profileData` with the `isBraintreeFastlaneProfileData` checking to avoid redundant `if`

Reply:
Good point! Fixed